arrow/src/util/bit_util.rs

```diff
@@ -107,6 +107,40 @@ pub fn ceil(value: usize, divisor: usize) -> usize {
     }
 }
 
+#[derive(Debug)]
+pub struct BitsIter<'a> {
+    bytes: &'a [u8],
+    offset: usize,
+    end_offset: usize,
+}
+
+impl<'a> BitsIter<'a> {
+    pub fn new(bytes: &'a [u8], offset: usize, len: usize) -> BitsIter<'a> {
+        let end_offset = offset + len;
+        if end_offset < offset || end_offset.div_ceil(8) > bytes.len() {
+            panic!("BitsIter::new called with invalid offset or len");
```

Comment:
I would suggest adding the current offset and len values, as well as the len of bytes slice to this message. This will probably never be needed, but if an logic error occurs somewhere and we see this panic in the logs, these values can help us figure out what's going on

Reply:
I agree, I added that info.